### PR TITLE
doesNotHaveElementsOfType

### LINKED ISSUE #962: doesNotHaveElementsOfType
#### Summary

Similiar to https://github.com/joel-costigliola/assertj-core/issues/202, I suggest adding `doesNotHaveElementsOfType(Class<*>)`

#### Example

```kotlin
assertThat(list).doesNotHaveElementsOfType(MyClass::class.java)
```

#### Java 8 specific ?

NO

* YES : create Pull Request from the `master` branch
* NO  : create Pull Request from the `2.x` branch

### PR DESCRIPTION
**Pull request by epeee**
#### Check List:
* Fixes #962 
* Unit tests : YES
* Javadoc with a code example (API only) : YES

### CODE REVIEW

File: src/main/java/org/assertj/core/api/AbstractIterableAssert.java [hunk 1]
```diff
@@ -536,6 +536,35 @@ public SELF hasOnlyElementsOfType(Class<?> expectedType) {
...
+   * assertThat(numbers).doesNotHaveElementsOfType(Integer.class);</code></pre>
+   *
+   * @param notExpectedType the not expected type.
+   * @return this assertion object.
+   * @throws NullPointerException if the given type is {@code null}.
+   * @throws AssertionError if one element is of the not expected type.
+   * @since 2.9.0 / 3.9.0
+   */
+  @Override
+  public SELF doesNotHaveElementsOfType(Class<?> notExpectedType) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractIterableAssert.java]: rename to `doesNotHaveAnyElementsOfType` (sorry late comment on #962)

File: src/main/java/org/assertj/core/api/AbstractIterableAssert.java [hunk 2]
```diff
@@ -536,6 +536,35 @@ public SELF hasOnlyElementsOfType(Class<?> expectedType) {
...
+   * <p>
+   * Example:
+   * <pre><code class='java'> List&lt;Number&gt; numbers = new ArrayList&lt;Number&gt;();
+   * numbers.add(1);
+   * numbers.add(2);
+   * numbers.add(3);
+   *
+   * // successful assertions:
+   * assertThat(numbers).doesNotHaveElementsOfType(Long.class);
+   * assertThat(numbers).doesNotHaveElementsOfType(Float.class);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractIterableAssert.java]: chain assertions to make the example lighter:
```
assertThat(numbers).doesNotHaveElementsOfType(Long.class)
                   .doesNotHaveElementsOfType(Float.class);
```

File: src/main/java/org/assertj/core/error/ShouldNotHaveElementsOfType.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Creates an error message indicating that a group have an element of the given type.
+ */
+public class ShouldNotHaveElementsOfType extends BasicErrorMessageFactory {
+
+  private ShouldNotHaveElementsOfType(Object actual, Class<?> notExpectedType, Class<?> notExpectedTypeOccurred) {
+    super("%n" +
+     "Expecting:%n" +
+     "  <%s>%n" +
+     "to not have elements of type:%n" +
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldNotHaveElementsOfType.java]: `"to not have any elements of the following types"`

Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldNotHaveElementsOfType.java]: ideally have a version when only one type was given and another when multiple were.

**Comment by epeee** [comment]
@joel-costigliola thx for your feedback! I just updated the pr.

**Comment by joel-costigliola** [comment]
Thanks @epeee ! I've just enhanced the error message by showing the unexpected types and the corresponding element values.

**Pull request closed without merging**